### ISSUE TITLE
How to load model trained by myself to test_models.py 

### ISSUE THREAD

**Issue by chenyihang1993** [mit-han-lab/temporal-shift-module#17]
I used this command to train the TSM model:
```
# You should get TSM_kinetics_RGB_resnet50_shift8_blockres_avg_segment8_e50.pth
python main.py kinetics RGB \
     --arch resnet50 --num_segments 8 \
     --gd 20 --lr 0.02 --wd 1e-4 --lr_steps 20 40 --epochs 50 \
     --batch-size 128 -j 16 --dropout 0.5 --consensus_type=avg --eval-freq=1 \
     --shift --shift_div=8 --shift_place=blockres --npb
```
And I got a ckpt.best.pth.tar and ckpt.pth.tar, likely including the model parameter, the model structure information, but the test_models.py only need the parameter. I tried to save the model parameter in ckpt.pth.tar and deleted the lines in test_models.py:
```
    # base_dict = {('base_model.' + k).replace('base_model.fc', 'new_fc'): v for k, v in list(checkpoint.items())}
    base_dict = {'.'.join(k.split('.')[1:]): v for k, v in list(checkpoint.items())}
    replace_dict = {'base_model.classifier.weight': 'new_fc.weight',
                    'base_model.classifier.bias': 'new_fc.bias',
                    }
    for k, v in replace_dict.items():
        if k in base_dict:
            base_dict[v] = base_dict.pop(k)

    net.load_state_dict(base_dict)
``` 
However, I got very low accuracy. Please tell me how to load the parameter rightly. THX.

**Comment by tonylins** [comment]
Hi, how did you run the script to test the performance of the model? the `test_models.py` file would assume you pass the name of the checkpoint to the file so that it can get the configurations like backbone, whether to use temporal shift, etc.

**Comment by chenyihang1993** [comment]
I renamed the model wights file of  ```ckpt.pth.tar``` as ```TSM_kinetics_RGB_resnet50_shift8_blockres_avg_segment8_e50.pth```, then run the script:
```
python test_models.py kinetics \
    --weights=pretrained/TSM_kinetics_RGB_resnet50_shift8_blockres_avg_segment8_e50.pth \
    --test_segments=8 --test_crops=1 \
    --batch_size=64
```

**Comment by tonylins** [comment]
Did you get a reasonable performance now?

**Comment by chenyihang1993** [comment]
Yes. THX

**Comment by ucasiggcas** [comment]
hi,dear
could you please the txt file or tell me where I can find it ?
thx
**kinetics/labels/val_videofolder.txt**

https://github.com/mit-han-lab/temporal-shift-module/issues/42
